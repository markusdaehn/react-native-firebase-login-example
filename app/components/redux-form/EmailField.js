import React, { Component } from 'react';
import { Field } from 'redux-form';
import Input from './Input';

export default class EmailField extends Component {
  constructor(props) {
    super(props);
  }

  focus() {
    this.field.getRenderedComponent().focus();
  }

  render() {
    return (
      <Field
        name='email'
        ref={(field) => this.field = field}
        withRef
        placeholder='Your email address'
        keyboardType="email-address"
        autoCorrect={false}
        autoCapitalize={'none'}
        component={Input}
        {...this.props}
      />
    );
  }
}
