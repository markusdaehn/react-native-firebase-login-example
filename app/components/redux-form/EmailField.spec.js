import 'react-native';
import React from 'react';
import EmailField from './EmailField';
import { shallow } from 'enzyme';

describe('EmailField', function () {
  let email;
  describe('when rendered with no props', function () {
    beforeEach(function () {
      email = shallow(<EmailField />)
    });

    it('should match snapshot with default props', function () {
      expect(email).toMatchSnapshot();
    });
  });
  describe('when rendered with name and placeholder properties set', function () {
    beforeEach(function () {
      email = shallow(<EmailField name='email-field-name' placeholder='email-field-placeholder'  />)
    });

    it('should match snapshot with name and placeholder set to passed in props', function () {
      expect(email).toMatchSnapshot();
    });
  });
});
