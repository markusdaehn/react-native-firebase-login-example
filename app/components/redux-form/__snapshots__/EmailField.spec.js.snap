// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailField when rendered with name and placeholder properties set should match snapshot with name and placeholder set to passed in props 1`] = `
<Field
  autoCapitalize="none"
  autoCorrect={false}
  component={[Function]}
  keyboardType="email-address"
  name="email-field-name"
  placeholder="email-field-placeholder"
  withRef={true}
/>
`;

exports[`EmailField when rendered with no props should match snapshot with default props 1`] = `
<Field
  autoCapitalize="none"
  autoCorrect={false}
  component={[Function]}
  keyboardType="email-address"
  name="email"
  placeholder="Your email address"
  withRef={true}
/>
`;
